Content.Shared/Turrets/SharedDeployableTurretSystem.cs

```diff
@@ -0,0 +1,163 @@
+    {
+        if (!ent.Comp.Enabled || args.Cancelled)
+            return;
+
+        _popup.PopupClient(Loc.GetString("deployable-turret-component-cannot-access-wires"), ent, args.User);
+
+        args.Cancelled = true;
+    }
+
+    public bool TryToggleState(Entity<DeployableTurretComponent> ent, EntityUid? user = null)
+    {
+        return TrySetState(ent, !ent.Comp.Enabled, user);
+    }
+
+    public bool TrySetState(Entity<DeployableTurretComponent> ent, bool enabled, EntityUid? user = null)
+    {
+        if (enabled && ent.Comp.CurrentState == DeployableTurretState.Broken)
+        {
+            if (user != null)
+                _popup.PopupClient(Loc.GetString("deployable-turret-component-is-broken"), ent, user.Value);
+
+            return false;
+        }
+
+        if (enabled && !HasAmmo(ent))
+        {
+            if (user != null)
+                _popup.PopupClient(Loc.GetString("deployable-turret-component-no-ammo"), ent, user.Value);
+
+            return false;
+        }
+
+        SetState(ent, enabled, user);
+
+        return true;
+    }
+
+    protected virtual void SetState(Entity<DeployableTurretComponent> ent, bool enabled, EntityUid? user = null)
+    {
+        if (ent.Comp.Enabled == enabled)
+            return;
+
+        // Hide the wires panel UI on activation
+        if (enabled && TryComp<WiresPanelComponent>(ent, out var wires) && wires.Open)
+        {
+            _wires.TogglePanel(ent, wires, false);
+            _audio.PlayPredicted(wires.ScrewdriverCloseSound, ent, user);
+        }
+
+        // Determine how much time is remaining in the current animation and the one next in queue
+        var animTimeRemaining = MathF.Max((float)(ent.Comp.AnimationCompletionTime - _timing.CurTime).TotalSeconds, 0f);
+        var animTimeNext = enabled ? ent.Comp.DeploymentLength : ent.Comp.RetractionLength;
+
+        ent.Comp.AnimationCompletionTime = _timing.CurTime + TimeSpan.FromSeconds(animTimeNext + animTimeRemaining);
+
+        // Change the turret's damage modifiers
+        if (TryComp<DamageableComponent>(ent, out var damageable))
+        {
+            var damageSetID = enabled ? ent.Comp.DeployedDamageModifierSetId : ent.Comp.RetractedDamageModifierSetId;
+            _damageable.SetDamageModifierSetId(ent, damageSetID, damageable);
+        }
+
+        // Change the turret's fixtures
+        if (ent.Comp.DeployedFixture != null &&
+            TryComp(ent, out FixturesComponent? fixtures) &&
+            fixtures.Fixtures.TryGetValue(ent.Comp.DeployedFixture, out var fixture))
+        {
+            _physics.SetHard(ent, fixture, enabled);
+        }
+
+        var msg = enabled ? "deployable-turret-component-activating" : "deployable-turret-component-deactivating";
+        _popup.PopupClient(Loc.GetString(msg), ent, user);
+
+        ent.Comp.Enabled = enabled;
+        Dirty(ent);
+    }
```

Comment:
If using delta fields need to call dirtyfield instead for all calls. ATM the state is so small it's not worth it but if we want it to be editable by admins then more of the fields would need autonetworkedfield for it.